hack/generate/samples/ansible/generate.go

```diff
@@ -0,0 +1,191 @@
+
+	gen := sample.NewGenerator(
+		sample.WithNoWebhook(),
+	)
+
+	err = gen.GenerateSamples(ansibleMemcached)
+	pkg.CheckError("generating ansible samples", err)
+
+	ImplementMemcached(ansibleMemcached, fmt.Sprintf("%s-%s", bundleImageBase, ansibleMemcached.Name()))
+	return []sample.Sample{ansibleMemcached}
+}
+
+// GenerateMoleculeSample will call all actions to create the directory and generate the sample
+// The Context to run the samples are not the same in the e2e test. In this way, note that it should NOT
+// be called in the e2e tests since it will call the Prepare() to set the sample context and generate the files
+// in the testdata directory. The e2e tests only ought to use the Run() method with the TestContext.
+func GenerateMoleculeSample(binaryPath, samplesPath string) {
+	ansibleCC := command.NewGenericCommandContext(
+		command.WithEnv("GO111MODULE=on"),
+		command.WithDir(filepath.Join(samplesPath, "")),
+	)
+
+	ansibleMoleculeMemcached := sample.NewGenericSample(
+		sample.WithBinary(binaryPath),
+		sample.WithCommandContext(ansibleCC),
+		sample.WithDomain("example.com"),
+		sample.WithGvk(
+			memcachedGVK,
+			schema.GroupVersionKind{
+				Group:   memcachedGVK.Group,
+				Version: memcachedGVK.Version,
+				Kind:    "Foo",
+			},
+			schema.GroupVersionKind{
+				Group:   memcachedGVK.Group,
+				Version: memcachedGVK.Version,
+				Kind:    "Memfin",
+			},
+		),
+		sample.WithPlugins("ansible"),
+		sample.WithExtraApiOptions("--generate-role", "--generate-playbook"),
+		sample.WithName("memcached-molecule-operator"),
+	)
+
+	addIgnore := sample.NewGenericSample(
+		sample.WithBinary(ansibleMoleculeMemcached.Binary()),
+		sample.WithCommandContext(ansibleMoleculeMemcached.CommandContext()),
+		sample.WithName(ansibleMoleculeMemcached.Name()),
+		sample.WithPlugins("ansible"),
+		sample.WithGvk(schema.GroupVersionKind{
+			Group:   "ignore",
+			Version: "v1",
+			Kind:    "Secret",
+		}),
+		sample.WithExtraApiOptions("--generate-role"),
+	)
+
+	// remove sample directory if it already exists
+	err := os.RemoveAll(ansibleMoleculeMemcached.Dir())
+	pkg.CheckError("attempting to remove sample dir", err)
+
+	gen := sample.NewGenerator(
+		sample.WithNoWebhook(),
+	)
+
+	err = gen.GenerateSamples(ansibleMoleculeMemcached)
+	pkg.CheckError("generating ansible molecule sample", err)
+
+	log.Infof("enabling multigroup support")
+	err = e2e.AllowProjectBeMultiGroup(ansibleMoleculeMemcached)
+	pkg.CheckError("updating PROJECT file", err)
+
+	ignoreGen := sample.NewGenerator(sample.WithNoInit(), sample.WithNoWebhook())
+	err = ignoreGen.GenerateSamples(addIgnore)
+	pkg.CheckError("generating ansible molecule sample - ignore", err)
+
+	ImplementMemcached(ansibleMoleculeMemcached, fmt.Sprintf("%s-%s", bundleImageBase, ansibleMoleculeMemcached.Name()))
+
+	ImplementMemcachedMolecule(ansibleMoleculeMemcached, fmt.Sprintf("%s-%s", bundleImageBase, ansibleMoleculeMemcached.Name()))
+}
```

Comment:
Note that the sample result should be the same.

Reply:
The samples should all be the same, I just noticed that I kept some test samples in my commits so I will remove them and that should create a much smaller PR.